bc_obps/service/user_profile_service.py

```diff
@@ -16,5 +17,5 @@ def create_user_profile(cls, user_guid: UUID, identity_provider: str, user_data:
             else AppRole.objects.get(role_name="cas_pending"),
             IdPs.BCEIDBUSINESS.value: AppRole.objects.get(role_name="industry_user"),
         }
-        role: AppRole = role_mapping.get(identity_provider)  # type: ignore[assignment] # we know this will not be None
+        role: AppRole = role_mapping.get(identity_provider, None)  # type: ignore[arg-type] # we know this will not be None
```

Comment:
I got mypy fails on this, okay to change the ignore?

Reply:
`None` is the default value anyways, this has no effect except triggering a different linter error.
A way of fixing this would be to make the signature of the method take an `IdPs` object instead of the string

Reply:
The mypy issue seemed to be a mismatch between the type Optional[AppRole] and AppRole